src/components/Firestore/DocumentEditor/index.tsx

```diff
@@ -217,10 +232,18 @@ const FieldEditor: React.FC<{
               <div className="DocumentEditor-ArrayEntry" key={c.uuid}>
                 <div className="DocumentEditor-ArrayEntryMetadata">
                   <div className="DocumentEditor-ArrayIndex">
-                    <Field value={index} label="Index" disabled />
+                    <Field
+                      value={index + startingIndex}
+                      label="Index"
+                      disabled
+                    />
                   </div>
                   <span className="Document-TypeSymbol">=</span>
-                  <ChildTypeSelect uuid={c.valueId} isRtdb={isRtdb} />
+                  <ChildTypeSelect
+                    uuid={c.valueId}
+                    isRtdb={isRtdb}
+                    blacklist={!supportNestedArrays ? [FieldType.ARRAY] : []}
```

Comment:
use const arrays since these will never change, and it will hopefully prevent a mysterious re-render later.

Reply:
Done